lib/propmangle.js

```diff
@@ -100,6 +101,13 @@ function mangle_properties(ast, options) {
 
     var regex = options.regex;
     var ignore_quoted = options.ignore_quoted;
+    
+    // note debug is either false (disabled), or a string of the debug cache name to use (enabled).
+    // note debug may be enabled as an empty string, which is falsey.
+    var debug = (options.debug !== false);
+    var debug_cache_name;
+    if (debug)
```

Comment:
Minor stylistic nit: either write a full block, or have the consequent on the same line.

In this case, you could just add it to the initialization as well (`var debug_cache_name = debug ? options.debug : null`)